Source/AliveLibAO/Paramite.cpp

```diff
@@ -1211,9 +1223,167 @@ void Paramite::State_2_Walking_44B9E0()
+        {
+            field_FC_current_motion = eParamiteStates::State_18_RunningAttack_44D5D0;
+            field_FE_next_state = -1;
+
+            if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+            {
+                field_B4_velx = -(ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(4));
+            }
+            else
+            {
+                field_B4_velx = (ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(4));
+            }
+        }
+
+        if (field_FE_next_state == eParamiteStates::State_6_Hop_44CB20)
+        {
+            VOnTrapDoorOpen();
+
+            field_E8_LastLineYPos = field_AC_ypos;
+            field_FC_current_motion = eParamiteStates::State_6_Hop_44CB20;
+            field_FE_next_state = -1;
+            field_B8_vely = -FP_FromDouble(7.2) * field_BC_sprite_scale;
+            field_AC_ypos += field_B8_vely;
+            field_F4_pLine = nullptr;
+
+            if (field_142)
+            {
+                MapFollowMe_401D30(1);
+                field_142 = 1;
+            }
+
+            SetMusic();
+        }
+
+        if (!field_142)
+        {
+            MapFollowMe_401D30(1);
+            field_142 = 1;
+        }
+
+        SetMusic();
+    }
+
+    if (field_10_anim.field_92_current_frame == 10)
+    {
+        Sfx_44DBB0(3u);
+        Abe_SFX_2_42A220(6, 50, 600, 0);
+
+        if (field_FE_next_state == eParamiteStates::State_2_Walking_44B9E0)
+        {
+            field_FC_current_motion = eParamiteStates::State_8_WalkRunTransition_44C790;
+            field_FE_next_state = -1;
+        }
+
+        if (!field_FE_next_state
+            || field_FE_next_state == eParamiteStates::State_5_Turn_44C8E0
+            || field_FE_next_state == eParamiteStates::State_23_Eating_44B970
+            || field_FE_next_state == eParamiteStates::State_25_Death_44DB90
+            || field_FE_next_state == eParamiteStates::State_15_Hiss_44D300)
+        {
+            field_FC_current_motion = eParamiteStates::State_11_RunEnd_44C620;
+        }
+
+        if (field_FE_next_state == eParamiteStates::State_18_RunningAttack_44D5D0)
+        {
+            field_FC_current_motion = eParamiteStates::State_18_RunningAttack_44D5D0;
+            field_FE_next_state = -1;
+
+            if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+            {
+                field_B4_velx = -(ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(4));
+            }
+            else
+            {
+                field_B4_velx = (ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(4));
+            }
+        }
+
+        if (field_FE_next_state == eParamiteStates::State_6_Hop_44CB20)
+        {
+            ToHop_44B660();
+        }
+
+        if (!field_142)
+        {
+            MapFollowMe_401D30(1);
+            field_142 = 1;
+        }
+
+        SetMusic();
```

Comment:
check this, was there a missing return here? otherwise its gonna overwrite field_142 again and call set music again